src/components/intern-app.html

```diff
@@ -44,6 +44,41 @@ <h1>Intern App</h1>
           }
         }
       }
+      constructor() {
+        super();
+        this.boundNewUser = this.newUser.bind(this);
+      }
+
+      connectedCallback() {
+        super.connectedCallback();
+        window.addEventListener("newUser", this.boundNewUser);   
+      }
+
+      disconnectedCallback() {
+        super.disconnectedCallback();
+        window.removeEventListener("newUser", this.boundNewUser); 
+      }
+
+      newUser(event) {
+        var newUserProfile = Object.assign({},event.detail);
+        newUserProfile.id = this.genId();
+        this.pushToArray(this.userProfiles, newUserProfile);
+      }
+
+      pushToArray(userProfiles, newUserProfile) {
+        var existingIds = userProfiles.map((newUserProfile) => newUserProfile.id);
+        if (!existingIds.includes(newUserProfile.id)) {
```

Comment:
I think you could skip the need for a .map and a .includes if you used .find with a function as your argument

```js
userProfiles.find(profile => {
  // some stuff in here
}) 
```



https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/find